repository-basex/src/main/java/org/lfenergy/compas/scl/data/repository/CompasDataBaseXRepository.java

```diff
@@ -0,0 +1,110 @@
+// SPDX-FileCopyrightText: 2021 Alliander N.V.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.scl.data.repository;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.lfenergy.compas.commons.MarshallerWrapper;
+import org.lfenergy.compas.scl.SCL;
+import org.lfenergy.compas.scl.data.basex.BaseXClient;
+import org.lfenergy.compas.scl.data.model.SclType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+
+@ApplicationScoped
+public class CompasDataBaseXRepository implements CompasDataRepository {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CompasDataBaseXRepository.class);
+
+    private final MarshallerWrapper marshallerWrapper;
+
+    private final String baseXHost;
+    private final Integer baseXPort;
+    private final String baseXUsername;
+    private final String baseXPassword;
+
+    @Inject
+    public CompasDataBaseXRepository(@ConfigProperty(name = "basex.host") String baseXHost,
+                                     @ConfigProperty(name = "basex.port") Integer baseXPort,
+                                     @ConfigProperty(name = "basex.username") String baseXUsername,
+                                     @ConfigProperty(name = "basex.password") String baseXPassword
+                                    ) throws Exception {
+        this.baseXHost = baseXHost;
+        this.baseXPort = baseXPort;
+        this.baseXUsername = baseXUsername;
+        this.baseXPassword = baseXPassword;
+
+        this.marshallerWrapper = new MarshallerWrapper.Builder().build();
+
+        // At startup create all needed databases.
+        for (SclType type: SclType.values()) {
+            executeCommand((client) -> {
+                client.execute("CREATE DB " + type.name());
+                return true;
+            });
+        }
+    }
+
+    @Override
+    public SCL findSCLByUUID(SclType type, UUID uuid) {
+        String result = executeQuery(type, "doc('".concat(type.name()).concat("/").concat(uuid.toString()).concat(".xml')"));
+        return marshallerWrapper.unmarshall(result.getBytes(StandardCharsets.UTF_8));
+    }
+
+    @Override
+    public UUID create(SclType type, SCL scl) {
+        UUID uuid = UUID.randomUUID();
+        InputStream inputStream = new ByteArrayInputStream(marshallerWrapper.marshall(scl).getBytes(StandardCharsets.UTF_8));
+        executeCommand((client) -> {
+            client.execute("OPEN ".concat(type.name()));
```

Comment:
OPEN/DELETE/CLOSE/etc.. command can be separate commands instead of using the same strings each time in the execute function.

Reply:
Edit: I now see the executeCommand function. I should have looked further.
It's creating a database client every time the execute function is executed so I'm not sure if it's handy to use this every time.
What is a good solution here?